src/backend/distributed/executor/multi_executor.c

```diff
@@ -617,12 +617,22 @@ IsLocalReferenceTableJoinPlan(PlannedStmt *plan)
 		return false;
 	}
 
-	foreach(oidCell, plan->relationOids)
+	foreach(rangeTableCell, plan->rtable)
```

Comment:
Does this contain RTEs for the entire query? (incl. CTEs, subqueries)

Reply:
Yes, standard_planner stores the flattened rte lists in plannedStmt->rtable. `set_plan_references` concatenates all these lists in `glob->finalrtable`, and then standard_planner does `plannedStmt->rtable = glob->finalrtable`.

Added a comment.